### PR TITLE
Handle JWTs with missing information

### PR DESCRIPTION
**Pull request by mrichards42**
These should return the same response as an invalid JWT (403).

Working on getting rid of junk errors ⚡️

### CODE REVIEW

File: src/turbovote_http_api/handlers/election_notifications.clj [hunk 1]
```diff
@@ -37,12 +37,16 @@
...
     (fn [ctx]
       (let [unsigned-token (utils/validate-token
                             (get-in ctx [:request :path-params :user-token]))
-            user-id (java.util.UUID/fromString (:user-id unsigned-token))]
-        (if (not= (::utils/status unsigned-token) :error)
+            user-id (try
+                      (java.util.UUID/fromString (:user-id unsigned-token))
+                      (catch IllegalArgumentException _
+                        nil))]
+        (if (and user-id (not= (::utils/status unsigned-token) :error))
```
Comment (dzcode) [inline src/turbovote_http_api/handlers/election_notifications.clj]: Why would a missing user-id be a 403 and not a 400?

Comment (mrichards42) [reply src/turbovote_http_api/handlers/election_notifications.clj]: b/c this is supposed to be an endpoint that requires some sort of authorization

File: src/turbovote_http_api/handlers/election_notifications.clj [hunk 2]
```diff
@@ -37,17 +37,21 @@
...
+                              java.util.UUID/fromString)
+                      (catch IllegalArgumentException _
+                        nil))]
+        (if (and user-id (not= (::utils/status unsigned-token) :error))
           (-> (assoc-in ctx [:request :path-params :user-id] user-id)
               (assoc-in  [:request :path-params :mediums] #{:email})
               (update-in [:request :path-params] dissoc :user-token))
-          (assoc-in ctx [:response :status]
-                    403))))}))
+          (utils/assoc-response ctx 403 "Not authorized"))))}))
```
Comment (mrichards42) [inline src/turbovote_http_api/handlers/election_notifications.clj]: A response with just a `:status` key doesn't actually do anything -- it has to be an actual ring response. So this interceptor was actually just letting the request pass through unchanged when it was invalid.

File: src/turbovote_http_api/handlers/election_notifications.clj [hunk 3]
```diff
@@ -37,17 +37,21 @@
...
+   {:name ::validate-token
+    :enter
     (fn [ctx]
       (let [unsigned-token (utils/validate-token
                             (get-in ctx [:request :path-params :user-token]))
-            user-id (java.util.UUID/fromString (:user-id unsigned-token))]
-        (if (not= (::utils/status unsigned-token) :error)
+            user-id (try
+                      (some-> (:user-id unsigned-token)
+                              java.util.UUID/fromString)
```
Comment (mrichards42) [inline src/turbovote_http_api/handlers/election_notifications.clj]: `UUID/fromString` actually throws `NullPointerException` when given `nil`, so safeguarding against this.

File: src/turbovote_http_api/handlers/election_notifications.clj (hunk 2 above)
Comment (dzcode) [reply src/turbovote_http_api/handlers/election_notifications.clj]: Looks like `utils/assoc-unauthorized` does this line

Comment (mrichards42) [reply src/turbovote_http_api/handlers/election_notifications.clj]: 👍 that sounds like a good refactor to me -- done

**Pull request merged**